src/diffusers/models/attention_processor.py

```diff
@@ -310,19 +310,16 @@ def set_attention_slice(self, slice_size):
 
         self.set_processor(processor)
 
-    def set_processor(self, processor: "AttnProcessor"):
-        if (
-            hasattr(self, "processor")
-            and not isinstance(processor, LORA_ATTENTION_PROCESSORS)
-            and self.to_q.lora_layer is not None
-        ):
+    def set_processor(self, processor: "AttnProcessor", _remove_lora=False):
+        if hasattr(self, "processor") and _remove_lora and self.to_q.lora_layer is not None:
```

Comment:
`self.to_q.lora_layer is not None` but is it a tight check, though? What happens if someone decides to not apply LoRA to the attention layers?

Reply:
Lora can only be removed if `self.to_q.lora_layer` is not None, so I think this is ok